src/Units/UnitContext.java

```diff
@@ -87,6 +106,8 @@ public void setPath(GamePath pPath)
   {
     path = pPath;
     coord = path.getEndCoord();
+    if( null != map && null != coord )
+      setEnvironment(map.getEnvironment(coord));
```

Comment:
Are these expected to ever be null? Should this not just be an error condition?

Reply:
Yes.
Them not being present is not an error.
If I want to contemplate a unit type outside of the context of a specific unit, then it feels pretty meaningless to include the map unless we care about the terrain for some reason.